client/my-sites/upgrades/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -0,0 +1,164 @@
+import {
+	get,
+	omit,
+} from 'lodash';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import SiteSelector from 'components/site-selector';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSites } from 'state/selectors';
+import Header from 'my-sites/upgrades/domain-management/components/header';
+import Main from 'components/main';
+import paths from 'my-sites/upgrades/paths';
+import { getSelectedDomain } from 'lib/domains';
+import NonOwnerCard from 'my-sites/upgrades/domain-management/components/domain/non-owner-card';
+import DomainMainPlaceholder from 'my-sites/upgrades/domain-management/components/domain/main-placeholder';
+import SectionHeader from 'components/section-header';
+import TransferConfirmationDialog from './confirmation-dialog';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class TransferToOtherSite extends React.Component {
+
+	static propTypes = {
+		selectedDomainName: React.PropTypes.string.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		currentUser: React.PropTypes.object.isRequired
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			targetSiteSlug: '',
+			showConfirmationDialog: false,
+			disableDialogButtons: false
+		};
+	}
+
+	isDataReady() {
+		return this.props.domains.hasLoadedFromServer;
+	}
+
+	checkSiteEligibility = ( site ) => {
+		return site.capabilities.manage_options &&
+			! site.jetpack &&
+			! get( site, 'options.is_domain_only', false ) &&
+			site.ID !== this.props.selectedSite.ID;
+	}
+
+	handleSiteSelect = ( targetSiteSlug ) => {
+		this.setState( {
+			targetSiteSlug,
+			showConfirmationDialog: true,
+		} );
+	}
+
+	handleConfirmTransfer = ( targetSite, closeDialog ) => {
+		const { selectedDomainName } = this.props,
+			targetSiteName = targetSite.name,
+			successMessage = this.props.translate(
+				'%(selectedDomainName)s has been transferred to site: %(targetSiteName)s',
+				{ args: { selectedDomainName, targetSiteName } } ),
+			defaultErrorMessage = this.props.translate(
+				'Failed to transfer %(selectedDomainName)s, please try again or contact support.', {
+					args: { selectedDomainName } } );
+		this.setState( { disableDialogButtons: true } );
+		wpcom.transferToSite( this.props.selectedSite.ID, this.props.selectedDomainName, targetSite.ID )
+			.then( () => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.successNotice( successMessage, { duration: 10000, isPersistent: true } );
+				closeDialog();
+				page( paths.domainManagementList( this.props.selectedSite.slug ) );
+			}, error => {
+				this.setState( { disableDialogButtons: false } );
+				this.props.errorNotice( error.message || defaultErrorMessage );
+				closeDialog();
+			} );
```

Comment:
Sorry for the late reply, @mcsf, but I had to brush up on my Redux... And I'm still puzzled why you think an action creator is needed here? I can't foresee another component using this action, so the DRY principle does not apply. Unless I'm missing something, in this case it seems it does not make sense to use one. I'd understand if it was a convention or in the guidelines, but I can't find anything about this being a rule.